okhttp/src/main/java/com/squareup/okhttp/Cache.java

```diff
@@ -142,12 +142,17 @@ public Cache(File directory, long maxSize) {
     cache = DiskLruCache.create(directory, VERSION, ENTRY_COUNT, maxSize);
   }
 
-  private static String urlToKey(Request request) {
+  private static String generateEntryKey(Request request) {
+    //If the request is for partial content, we want a unique key for the partial response
```

Comment:
Yeah.  Makes complete sense.  Thinking about it further, it's much more practical of a case that a request for 101-200 bytes proceed a request for 0 - 100 bytes.  I could see cache hit rates increase for cases of requesting a full body, then subsequently requesting partial bodies; but even that is probably something of an edge case.